domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkManagerInitializer.kt

```diff
@@ -0,0 +1,63 @@
+package org.oppia.android.domain.platformparameter.syncup
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequest
+import androidx.work.WorkManager
+import org.oppia.android.domain.oppialogger.ApplicationStartupListener
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Enqueues unique periodic work requests for fetching and caching latest platform parameter values
+ * from the remote service on application creation.
+ */
+@Singleton
+class PlatformParameterSyncUpWorkManagerInitializer @Inject constructor(
+  private val context: Context
+) : ApplicationStartupListener {
+
+  private val OPPIA_PLATFORM_PARAMETER_WORK_REQUEST_NAME = "OPPIA_PLATFORM_PARAMETER_WORK_REQUEST"
+
+  private val platformParameterSyncUpWorkerConstraints = Constraints.Builder()
+    .setRequiredNetworkType(NetworkType.CONNECTED)
+    .setRequiresBatteryNotLow(true)
+    .build()
+
+  private val workerTypeForSyncingPlatformParameters: Data = Data.Builder()
+    .putString(
+      PlatformParameterSyncUpWorker.WORKER_TYPE_KEY,
+      PlatformParameterSyncUpWorker.PLATFORM_PARAMETER_WORKER
+    )
+    .build()
+
+  private val workRequestForSyncingPlatformParameters = PeriodicWorkRequest
+    .Builder(PlatformParameterSyncUpWorker::class.java, 12, TimeUnit.HOURS)
```

Comment:
I am wondering if we should put the number 12 as a dagger constant so that we can change it easily

Reply:
Yeah it is a good idead but just to make confirm this it will mean introducing a new platform parameter in the codebase.